SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.cpp

```diff
@@ -0,0 +1,278 @@
+				if (Timer <= 0)
+				{
+					CheckComponents(DynamicReplicatedActor, 1, 0);
+				}
+			},
+			5.0f);
+
+		// Client 1 to get ownership of the dynamic actor - then check components
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Set owner to client 1"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					AController* PlayerController =
+						Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+					DynamicReplicatedActor->SetOwner(PlayerController);
+					FinishStep();
+				});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Same Server - Verify components"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					// FinishStep();
+					CheckComponents(DynamicReplicatedActor, 1, 1, 0);
+				}
+			},
+			5.0f);
+
+		// Client 2 to get ownership of the dynamic actor - then check components
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Set owner to client 2"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					AController* PlayerController =
+						Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 2)->GetOwner());
+					DynamicReplicatedActor->SetOwner(PlayerController);
+					FinishStep();
+				});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Same Server - Verify components"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					CheckComponents(DynamicReplicatedActor, 1, 2, 1);
+				}
+			},
+			5.0f);
+
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(1));
+	}
+
+	// Replicated Dynamic Actor Spawned On Different Server. Server 1 should have Authority on BeginPlay, Server 2 on Tick
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Different Server - Spawn"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			ASpatialComponentTestReplicatedActor* Actor =
+				GetWorld()->SpawnActor<ASpatialComponentTestReplicatedActor>(Server2Position, FRotator::ZeroRotator);
+			CrossServerSetDynamicReplicatedActor(Actor);
+			FinishStep();
+		});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Different Server - Verify components "
+				 "on Tick"),
+			FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					CheckComponentsCrossServer(DynamicReplicatedActor, 1, 2);
+				}
+			},
+			5.0f);
+
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(2));
+	}
+}
+
+void ASpatialComponentTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialComponentTest, DynamicReplicatedActor);
+}
+
+// Checks the number of components on the servers when an actor does not migrate
```

Comment:
Comment doesn't mention clients, but we do verify components on clients in addition to servers, might be worth adding to the comment?

Reply:
Updated comment